awx/network_ui/consumers.py

```diff
@@ -0,0 +1,281 @@
+    topology_data = transform_dict(dict(id='topology_id',
+                                        name='name',
+                                        panX='panX',
+                                        panY='panY',
+                                        scale='scale',
+                                        link_id_seq='link_id_seq',
+                                        device_id_seq='device_id_seq'), topology.__dict__)
+
+    message.reply_channel.send({"text": json.dumps(["Topology", topology_data])})
+    send_snapshot(message.reply_channel, topology_id)
+
+
+def send_snapshot(channel, topology_id):
+    interfaces = defaultdict(list)
+
+    for i in (Interface.objects
+              .filter(device__topology_id=topology_id)
+              .values()):
+        i = transform_dict(dict(cid='id',
+                                device_id='device_id',
+                                id='interface_id',
+                                name='name'), i)
+        interfaces[i['device_id']].append(i)
+    devices = list(Device.objects.filter(topology_id=topology_id).values())
+    devices = [transform_dict(dict(cid='id',
+                                   id='device_id',
+                                   device_type='device_type',
+                                   host_id='host_id',
+                                   name='name',
+                                   x='x',
+                                   y='y',
+                                   interface_id_seq='interface_id_seq'), x) for x in devices]
+    for device in devices:
+        device['interfaces'] = interfaces[device['device_id']]
+
+    links = [dict(id=x['cid'],
+                  name=x['name'],
+                  from_device_id=x['from_device__cid'],
+                  to_device_id=x['to_device__cid'],
+                  from_interface_id=x['from_interface__cid'],
+                  to_interface_id=x['to_interface__cid'])
+             for x in list(Link.objects
+                               .filter(Q(from_device__topology_id=topology_id) |
+                                       Q(to_device__topology_id=topology_id))
+                               .values('cid',
+                                       'name',
+                                       'from_device__cid',
+                                       'to_device__cid',
+                                       'from_interface__cid',
+                                       'to_interface__cid'))]
+    snapshot = dict(sender=0,
+                    devices=devices,
+                    links=links)
+    channel.send({"text": json.dumps(["Snapshot", snapshot])})
+
+
+@channel_session
+def ws_message(message):
+    # Send to all clients editing the topology
+    Group("topology-%s" % message.channel_session['topology_id']).send({"text": message['text']})
+    # Send to persistence handler
+    Channel('persistence').send({"text": message['text'],
```

Comment:
`networking_events`

Reply:
I removed this completely in favor of a function call.